static/js/containers/ProfilePage.js

```diff
@@ -40,9 +39,7 @@ class ProfilePage extends React.Component {
       dispatch(startProfileEdit());
     }
     return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
-      dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
-        dispatch(clearProfileEdit());
-      });
+      dispatch(saveProfile(SETTINGS.username, profile));
```

Comment:
I just made the change above locally, and if I have a validation error on the `personal` tab and click 'save and continue' on the privacy tab I see the issue you're talking about.